data/test_moodbot/unexpected_intent_policy_config.yml

```diff
@@ -0,0 +1,15 @@
+language: en
+
+pipeline:
+
+policies:
+  - name: MemoizationPolicy
+  - name: UnexpecTEDIntentPolicy
+    max_history: 5
+    epochs: 100
```

Comment:
do we really need `epochs: 100` for `UnexpecTED` and `TEDPolicy`? Seems its only used in tests where it's mocked anyway

Reply:
I had set the epochs to `1` but I think @alwx bumped it up to `100` again for some tests. @alwx Can you please comment?

Reply:
I had set the epochs to `1` but I think @alwx bumped it up to `100` again for some tests. @alwx Can you please comment?

Reply:
in fact, the tests fail when I set `epochs` to `1`

Reply:
Which ones?

Reply:
>  we force them to return action_unlikely_intent after a specified intent

Can you please mention the test names which fail? For those tests can't we monkey-patch this behaviour of forcing them to return `action_unlikely_intent` after a specified intent. As I mentioned earlier, relying on epochs to achieve deterministic behaviour from ML models will result in flaky tests.

Reply:
From what I can see the tests inside `test_test.py`, for example `test_action_unlikely_intent_1` will be flaky because they rely on deterministic predictions from all policies not just `UnexpecTEDIntentPolicy`. Even `TEDPolicy` will fail there if the pipeline is trained for only 1 epoch.

Reply:
Looking into how we can make these tests not rely on an actual trained agent.

Reply:
Being fixed here - https://github.com/RasaHQ/rasa/pull/9063